docs/DeveloperGuide.md

```diff
@@ -6,33 +6,181 @@
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+
+  1. Program should work on any operating system supporting Java 11 or above. 
+  2. Does not require an active connection to the Internet to use the application.
+  3. Program should be accessible to users with no prior programming experience.
+  4. Program data will be saved persistently. 
+  5. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse. 
+
 
 ## Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
-
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+## Instructions for Manual Testing
+### Launch
+1. Download the JAR file and move it into an empty folder.
+2. On a command line application, change the current working directory to the same folder as the JAR file and run the app using:\
+   `java -jar tp.jar`
+3. Expected: The app's welcome message is printed onto the terminal along with list of all commands.
+
+### Add 
+#### Add past expense
+1. Prerequisites:
+   - Ensure that a valid date is entered
+   - Date is entered in the format: yyyymmdd
+   - Ensure all fields are entered
+2. Test case: 
+   `add expense book $/100 d/20230101`\
+   Expected: Terminal shows choose category message. Enter a category number. Expense entry is added. Terminal shows successful message along with expenditure details (name, amount, date, category).
+   
+#### Add future expense
+1. Prerequisites:
+   - Ensure that a valid date is entered
+   - Date is entered in the format: yyyymmdd
+   - Ensure all fields are entered
+2. Test case:
+   `add future expense bill $/200 d/20230505`\
+   Expected: Terminal shows choose category message. Enter a category number. Expense entry is added. Terminal shows successful message along with expenditure details (name, date, category).
+
+#### Entering incomplete information
+Test case: `add expense book $/100` or `add expense book d/20220101` or `add expense book`
+Expected: Terminal shows error message.
+
+### Delete
+#### Delete past expense
+1. Prerequisites:
+   - Ensure that a valid expense id is entered
+   - Ensure all fields are entered
+2. Test case:
+   `delete expense id/1`\
+   Expected: Expense entry is deleted. Terminal shows successful message along with expenditure details (name, amount, date, category).
+
+#### Delete future expense
+1. Prerequisites:
+   - Ensure that a valid future expense id is entered
+   - Ensure all fields are entered
+2. Test case:
+   `delete future expense id/1`\
+   Expected: Expense entry is deleted. Terminal shows successful message along with expenditure details (name, date, category).
+
+#### Entering incomplete information
```

Comment:
It would be useful to get an example of the type of error message - is it a specific exception message? A screenshot would be useful to understand the expected outcome. More expected outcomes in general would be helpful.